File: app/src/main/java/org/dhis2hiv/usescases/jira/JiraIssueHolder.java
package org.dhis2hiv.usescases.jira;

import org.dhis2hiv.databinding.JiraIssueItemBinding;
import org.dhis2hiv.utils.jira.JiraIssue;
import org.dhis2hiv.utils.jira.OnJiraIssueClick;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

class JiraIssueHolder extends RecyclerView.ViewHolder {
    private final JiraIssueItemBinding binding;

    JiraIssueHolder(@NonNull JiraIssueItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(JiraIssue jiraIssue, OnJiraIssueClick listener) {
        binding.setIssue(jiraIssue);
        itemView.setOnClickListener(view -> listener.onJiraIssueClick(jiraIssue.getKey()));
    }
}

File: app/src/main/java/org/medhis2yes/usescases/jira/JiraIssueHolder.kt
package org.medhis2yes.usescases.jira

import androidx.recyclerview.widget.RecyclerView
import org.medhis2yes.data.jira.JiraIssue
import org.medhis2yes.databinding.JiraIssueItemBinding

class JiraIssueHolder(
    private val binding: JiraIssueItemBinding,
    private val onJiraIssueClick: (String) -> Unit
) : RecyclerView.ViewHolder(binding.root) {
    fun bind(jiraIssue: JiraIssue) {
        binding.apply {
            issue = jiraIssue
            jiraCard.setOnClickListener { onJiraIssueClick(jiraIssue.key) }
        }
    }
}
